### PR TITLE
Clean up the language-specific schema APIs.

### PR DESCRIPTION
**Pull request by pgavlin**
Rather than forcing consumers to deal with language-specific data
ad-hoc, add an API that allows all language-specific data to be decoded
up-front.

### CODE REVIEW

File: pkg/codegen/dotnet/importer.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// Copyright 2016-2018, Pulumi Corporation.
```
Comment (lblackstone) [inline pkg/codegen/dotnet/importer.go:1]: nit: Outdated copyright

File: pkg/codegen/dotnet/importer.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// CSharpPackageInfo represents the C# language-specific info for a package.
+type CSharpPackageInfo struct {
+	PackageReferences map[string]string `json:"packageReferences,omitempty"`
+	Namespaces        map[string]string `json:"namespaces,omitempty"`
+}
+
+// Importer implements schema.Language for .NET.
+var Importer schema.Language = importer(0)
+
+type importer int
```
Comment (lblackstone) [inline pkg/codegen/dotnet/importer.go:37]: Any reason to make this an `int` rather than an empty struct?

File: pkg/codegen/schema/schema.go [hunk 3]
```diff
@@ -284,7 +284,200 @@ type Package struct {
...
+	return nil
+}
+
+func importPropertyLanguages(property *Property, languages map[string]Language) error {
+	if property.DefaultValue != nil {
+		if err := importDefaultLanguages(property.DefaultValue, languages); err != nil {
+			return errors.Wrapf(err, "importing default value")
+		}
+	}
+
```
Comment (lblackstone) [inline pkg/codegen/schema/schema.go:337]: Should we also handle `ConstValue` here?

Comment (pgavlin) [reply pkg/codegen/schema/schema.go:337]: Ah, yes.

File: pkg/codegen/dotnet/importer.go (hunk 2 above)
Comment (pgavlin) [reply pkg/codegen/dotnet/importer.go:37]: I see `int` a bit more frequently for this sort of stateless interface.

File: pkg/codegen/schema/schema.go (hunk 3 above)
Comment (pgavlin) [reply pkg/codegen/schema/schema.go:337]: Oh, wait--we don't have language-specific info for consts. We'll need to if/when we do.

File: pkg/codegen/dotnet/importer.go (hunk 1 above)
Comment (pgavlin) [reply pkg/codegen/dotnet/importer.go]: Fixed

**Pull request merged**